pkg/trace/config/apply.go

```diff
@@ -382,6 +383,28 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		c.LogThrottling = false
 	}
 
+	if config.Datadog.GetBool("apm_config.internal_profiling.enabled") {
+		profilingSite := config.Datadog.GetString("internal_profiling.profile_dd_url")
+		if profilingSite == "" {
+			profilingSite = site
+		}
+
+		c.ProfilingSettings = &profiling.Settings{
+			Site: profilingSite,
+
+			// remaining configuration parameters use the top-level `internal_profiling` config
+			Period:               config.Datadog.GetDuration("internal_profiling.period"),
+			CPUDuration:          config.Datadog.GetDuration("internal_profiling.cpu_duration"),
+			MutexProfileFraction: config.Datadog.GetInt("internal_profiling.mutex_profile_fraction"),
+			BlockProfileRate:     config.Datadog.GetInt("internal_profiling.block_profile_rate"),
+			WithGoroutineProfile: config.Datadog.GetBool("internal_profiling.enable_goroutine_stacktraces"),
```

Comment:
This part seems like it could be common to all. `profiling.Start` could read it.

Reply:
It's different in different ways each place it's used, so there's no getting around doing it four times without changing the meaning of existing configs.  The best we can do is to isolate it and not pollute every AgentConfig with a dozen ProfilingXxx parameters.

Off the top of my head:
 * different way of calculating `site`
 * different config names for profiling-specific adjustments
 * different defaults